### PR TITLE
Graphics layer init

### PR DESCRIPTION
**Pull request by araml**
Part 1 of 2 in #8  

Shows a simple image in the middle of the screen. For this you need the stb_image library (NOT INCLUDED in the repo since we'll replace it with SDL_Image soon) download it from [here](https://github.com/nothings/stb/blob/master/stb_image.h) and drop it in src/

Also merged build scripts into one, now build.sh takes the string "Ninja" as input to specify a ninja build.

### CODE REVIEW

File: src/math/point.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#pragma once
+#ifndef POINT_H
+#define POINT_H
+
+#include <cmath>
+#include <ostream>
+
+using uchar = unsigned char;
```
Comment (dysoco) [inline src/math/point.h]: Can we use stdint.h and fixed-size types instead?

**Comment by dysoco** [comment]
Looks good to me but I don't know OpenGL so I'm gonna have to trust you.

Why did you use stb_image if you were planning to use SLD_Image later anyways?

File: src/main.cpp [hunk 2]
```diff
@@ -29,6 +46,11 @@ int main(int argc, char *argv[]) {
                 quit = true;
             }
         }
+
+        std::this_thread::sleep_for(milliseconds{10});
```
Comment (Saeuissimus) [inline src/main.cpp]: Why not use millisecond literals?

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,8 +1,5 @@
 cmake_minimum_required(VERSION 3.0)
 
-SET(CMAKE_C_COMPILER /usr/bin/clang)
-SET(CMAKE_CXX_COMPILER /usr/bin/clang++)
-
 project(optativa)
```
Comment (Saeuissimus) [inline CMakeLists.txt]: Moving these is after the `project` directive is actually a bad idea. See the relevant [FAQ](https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#method-3-avoid-use-set).

Comment (araml) [reply CMakeLists.txt]: OK, ahi lo arreglo.

File: src/math/point.h [hunk 4]
```diff
@@ -0,0 +1,81 @@
+#pragma once
+#ifndef POINT_H
```
Comment (dgarroDC) [inline src/math/point.h]: Why do you se `#pragma once` AND `#ifndef`? Please review this in all the project.

File: src/renderer/texture.h [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+class texture {
+public:
+    texture(const std::string &path, int width, int height);
+private:
+    void load_texture();
+    void init_buffers();
+    void create_quad(int screen_width, int screen_height);
+    void update_texture_info();
+    void bind_texture_data();
+    void generate_atlas_positions(int sprite_width, int sprite_height,
```
Comment (dgarroDC) [inline src/renderer/texture.h]: 🤔

File: src/renderer/renderer.cpp [hunk 6]
```diff
@@ -15,9 +16,16 @@ void renderer::clear() {
...
+    glBindFramebuffer(GL_FRAMEBUFFER, 0);
+    glViewport(0, 0, 640, 480);
+    SDL_GL_SwapWindow(window);
+}
 
+void renderer::draw(texture &tex) {
+    default_shader.use();
+    glBindTexture(GL_TEXTURE_2D, tex.tex);
+    glBindVertexArray(tex.vao);
+    glDrawElements(GL_TRIANGLES, 6, GL_UNSIGNED_INT, 0);
```
Comment (dgarroDC) [inline src/renderer/renderer.cpp]: Please considerer drawing with `glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);`(with the vertex in the right order) and getting rid of the ebo.

**Pull request merged**